wgsl/index.bs

```diff
@@ -347,6 +347,13 @@ A vector type is a <dfn dfn>numeric vector</dfn> type if its component type is a
                        These may only appear in specific contexts.<br>
 </table>
 
+An array element type must not be:
+
+* The [=void=] type.
```

Comment:
This is a bit of a laundry list.

I *could* have defined a new concept such as a "concrete" type.

However, I expect WGSL will soon have to support arrays of textures and arrays of samplers, so the category isn't going to be particularly stable.